cluster/provision_gateway.sh

```diff
@@ -0,0 +1,675 @@
+     fi
+     if [ "$ifmapup" == true ]; then
+       log_info_msg "IFMAP and Contrail-Control are up are ready for processing agent subscriptions"
+       break
+     fi
+     sleep 3
+    done
+  mv /tmp/contrail-vrouter-agent.manifest /etc/kubernetes/manifests
+  
+  # start docker 
+  img=$(docker images | grep vrouter | awk '{print $1}')
+  ver=$(docker images | grep vrouter | awk '{print $2}')
+  
+  docker run --privileged -d -P --name contrail-vrouter-agent --net="host" -t -i -e sysimage=/host -v /etc/contrail:/etc/contrail -v /var/log/contrail:/var/log/contrail $img:$ver /usr/bin/contrail-vrouter-agent
+
+  # start monitoring
+  wget -P /etc/contrail https://raw.githubusercontent.com/Juniper/contrail-kubernetes/$ockver/scripts/opencontrail-install/contrail_vr_agent_mon.sh
+  chmod +x /etc/contrail/contrail_vr_agent_mon.sh
+  cron="*/3 * * * * /etc/contrail/contrail_vr_agent_mon.sh  2>&1 | logger"
+  (crontab -u root -l; echo "$cron" ) | crontab -u root -
+}
+
+function ifup_vhost()
+{
+  ifup $VHOST
+  intf=$(vif --list | grep vhost | awk '{print $3}')
+  if [[ "$intf" == $VHOST ]]; then
+    log_info_msg "Vhost setup successfuly"
+  else
+    log_info_msg "Vhost setup - Error"
+  fi
+}
+
+function routeconfig()
+{
+    if isGceVM ; then
+       #configure point to point route
+       #asssuming the container is debian or ubuntu
+       naddr=$(getGceNetAddr)
+       defgw=$(ip route | grep default | awk '{print $3}')
+       route add -host $defgw $VHOST
+       route del -net $naddr dev $VHOST
+       hostroute="up route add -host $defgw dev $VHOST"
+       netdel="up route del -net $naddr dev $VHOST"
+       grep -q 'up route add -host' /etc/network/interfaces || sed -i "/gateway/a \\\t$hostroute\n\t$netdel" /etc/network/interfaces
+    fi
+}
+
+function verify_vhost_setup()
+{
+  sleep 3
+  status=$(ping -c 1 -w 1 -W 1 -n $OPENCONTRAIL_CONTROLLER_IP | grep packet | awk '{print $6}' | cut -c1)
+  if [ "$status" == 0 ]; then
+    log_info_msg "Vrouter kernel module and network successfuly setup"
+  else
+    log_info_msg "Vrouter kernel module and network - Error"
+  fi
+}
+
+function provision_vrouter()
+{
+  stderr="/tmp/stderr"
+  host=`hostname -s`
+
+  # check if contrail-api is up 60 times, for 5 mins
+  vr=''
+  for (( i=0; i<60; i++ ))
+    do
+     vr=$(curl -s http://$OPENCONTRAIL_CONTROLLER_IP:8082 | grep -ow "virtual-routers")
+     if [ ! -z $vr ]; then
+       break
+     fi
+     sleep 5
+    done
+  curl -s -X POST -H "Content-Type: application/json; charset=UTF-8" -d '{"virtual-router": {"parent_type": "global-system-config", "fq_name": ["default-global-system-config", "'$host'" ], "display_name": "'$host'", "virtual_router_ip_address": "'$MINION_OVERLAY_NET_IP'", "name": "'$host'"}}' http://$OPENCONTRAIL_CONTROLLER_IP:8082/virtual-routers 2> >( cat <() > $stderr)
```

Comment:
This shouldn't be done from the host. The master should invoke the python provision-vrouter script.